src/ledger/NetworkConfig.h

```diff
@@ -288,7 +289,7 @@ class SorobanNetworkConfig
     initializeGenesisLedgerForTesting(uint32_t genesisLedgerProtocol,
                                       AbstractLedgerTxn& ltx, Application& app);
 
-    void loadFromLedger(AbstractLedgerTxn& ltx, uint32_t configMaxProtocol,
+    void loadFromLedger(LedgerSnapshot const& ls, uint32_t configMaxProtocol,
```

Comment:
What’s the rationale for this change? loadFromLedger explicitly requires in-progress LedgerTxn because it has the most recent apply state (config is loaded _before_ we commit LCL and create a new snapshot). Adding support for bucket list snapshots seems like a footgun since it may be in stale state.

Reply:
The goal was to stop passing in `LedgerTxn` into what should be a read only function. Would `LedgerTxnReadOnly` work here instead?

Reply:
Looks like `LedgerTxnReadOnly` works, and that gives us exactly what we want.

Reply:
LedgerTxnReadOnly should work since it doesn’t support bucket list snapshots at all